python/ray/ml/checkpoint.py

```diff
@@ -1,35 +1,94 @@
-import abc
-from typing import Dict, Optional
+import shutil
+import tarfile
+import tempfile
+
+import cloudpickle as pickle
+import os
+from typing import Optional
 
 import ray
+from ray.rllib.utils import DeveloperAPI
+from ray.util.ml_utils.cloud import (
+    upload_to_bucket,
+    is_cloud_target,
+    download_from_bucket,
+)
+
+
+DICT_CHECKPOINT_FILE_NAME = "dict_checkpoint.pkl"
+
+
+class Checkpoint:
+    """Ray ML Checkpoint.
 
+    This implementation provides interfaces to translate between
+    different checkpoint storage locations: Local FS storage, remote
+    node FS storage, data object, and cloud storage location.
+    """
 
-class Checkpoint(abc.ABC):
-    """Checkpoint interface"""
+    @DeveloperAPI
+    def __init__(
```

Comment:
Some comments here? I see there are at least some assumptions about the format of uri, which one takes priority etc.

Reply:
I've aded this to the `from_uri` now as the constructor is private.